### ISSUE TITLE
Bug: Unexpected Exit Code with cucumberReporter When Setup Fails

### ISSUE THREAD

**Issue by pplancq** [vitalets/playwright-bdd#200]
**Given**

I have set up a setup-project to initialize authentication for my tests, following [Playwright's documentation here](https://playwright.dev/docs/auth)

**And**
I implemented the junit reporter via the function [cucumberReporter](https://vitalets.github.io/playwright-bdd/#/reporters/cucumber?id=junit)

**When**

If for some reason the setup fails, it does not launch the tests, which is expected. However, the exit code of Playwright is 0 instead of 1.

**Then**

This causes a CI/CD pipeline to not turn red, giving the false impression that everything is fine. [An example pipeline can be found here.](https://github.com/pplancq/playwright-bug-ci/actions/runs/10211299906/job/28252367739)
![image](https://github.com/user-attachments/assets/6a119b6a-cee6-46f3-8d75-95bb6ed3ab9f)

**But I expect**

I expect the exit code to be 1 when the setup fails, indicating an error.

**Isolated demo**

[pplancq/playwright-bug-ci](https://github.com/pplancq/playwright-bug-ci)

**Environment**

```
Playwright-bdd environment info:

platform: win32
node: v20.16.0
playwright-bdd: v7.1.2
@playwright/test: v1.45.3
@cucumber/cucumber: v
Playwright config file: playwright.config.js
```

```javascript
// playwright.config.js
const { defineConfig, devices } = require('@playwright/test');
const { defineBddConfig, cucumberReporter } = require('playwright-bdd');

const testDir = defineBddConfig({
  features: 'tests/features/**/*.feature',
  steps: 'tests/steps/**/*.steps.js',
});

module.exports = defineConfig({
  testDir,
  fullyParallel: true,
  forbidOnly: !!process.env.CI,
  retries: process.env.CI ? 2 : 0,
  workers: process.env.CI ? 1 : undefined,
  reporter: [
    cucumberReporter('junit', {
      outputFile: 'cucumber-report/report.xml',
      suiteName: 'my suite'
    }),
  ],
  use: {
    trace: 'on-first-retry',
  },
  projects: [
    {
      name: 'setup auth',
      testDir: './tests/',
      testMatch: /global\.setup\.ts/,
    },
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
        storageState: 'playwright/.auth/user.json'
      },
      dependencies: ['setup auth'],
    },
  ],
});
```

**Comment by vitalets** [comment]
Hi @pplancq , I will check.

**Comment by vitalets** [comment]
Fixed in **v7.2.1**.
Could you re-check?

**Comment by vitalets** [comment]
You are welcome!
